src/library/scala/concurrent/ExecutionContext.scala

```diff
@@ -139,7 +139,22 @@ object ExecutionContext {
    *
    * @return the global `ExecutionContext`
    */
-  def global: ExecutionContextExecutor = Implicits.global.asInstanceOf[ExecutionContextExecutor]
+  final def global: ExecutionContextExecutor = Implicits.global.asInstanceOf[ExecutionContextExecutor]
+
+  /**
+   * The `callingThread` `ExecutionContext` executes `Runnable`s on the calling `Thread`.
+   * Nested submissions will be trampolined to prevent uncontrolled stack space growth.
+   * Any `NonFatal` or `InterruptedException`s will be reported to the `defaultReporter`.
+   *
+   * It is advised not to call any blocking code in the `Runnable`s submitted to this `ExecutionContext`
```

Comment:
this warning honestly doesn't seem strong enough to me. is there any valid situation where you should call blocking code using this `ExecutionContext`?

Reply:
I personally lean towards something like "DO NOT call any blocking code ..."

Reply:
Great point, @NthPortal. I have updated the Scaladoc—what do you think of the new text?